### PR TITLE
tests for plugin module

### PR DESCRIPTION
**Pull request by chillipeper**
Here is a first attempt to add some coverage to the plugin module.

- messages and events get created with a timestamp so added freeze gun module, to mock datetime ocurrences 
- added event fixture to conftest
- added io_backend and analysis parametrized fixtures of available backends to conftest
- Bump up coverage to 60%

Reg test_say and test_reply, the only test added are the ones that have a return value.

I haven't figured out yet how to test or validate the `self.publish` method. 

Any comments, suggestions or corrections more than welcome :)

IMPORTANT: time chosen to freeze_time is actually Will's Birthday 😄

### CODE REVIEW

**Comment by chillipeper** [comment]
@rsalmond maybe you could take a look into this when you have some spare time? 

Any suggestions would be really appreciated

File: will/tests/test_plugin.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+
+@pytest.fixture
+def content():
+    return "This is a test content"
+
+
+def test__prepared_content(plugin, content):
+    assert plugin._prepared_content(
+        content, "", "") == re.sub(r'>\s+<', '><', content)
```
Comment (wontonst) [inline will/tests/test_plugin.py]: I think it makes more sense to assert it against a pinned output: "This is a test content"

Comment (wontonst) [reply will/tests/test_plugin.py]: also, I don't think prepared_content is used anywhere

File: will/tests/test_plugin.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+               'kwargs': {}})
+    # Package plugin.say in a tuple to be able to make an assertion
+    plugin_say = (plugin.say(content, message=original_message, package_for_scheduling=True))
+    assert plugin_say == ("message.outgoing.%s" % backend, e)
+
+
+@freeze_time(WILLS_BIRTHDAY)
+def test_reply_package_for_scheduling_is_true(plugin, content, message, event, io_backend):
+    original_message = message({"backend": io_backend})
+    incomming_event = event({"data": original_message})
```
[the remark this answers is not in the document]
Comment (chillipeper) [reply will/tests/test_plugin.py]: changed

File: will/tests/test_plugin.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    backend = plugin.get_backend(original_message, None)
+    e = event({'type': "reply",
+               'content': content,
+               'topic': "message.outgoing.%s" % backend,
+               'source_message': original_message,
+               'kwargs': {}})
+    plugin_reply = plugin.reply(incomming_event, content, package_for_scheduling=True)
```
[the remark this answers is not in the document]
Comment (chillipeper) [reply will/tests/test_plugin.py]: changed

File: will/tests/test_plugin.py (hunk 1 above)
Comment (chillipeper) [reply will/tests/test_plugin.py]: yeahh, didn't find it being used anywhere either.

Would it be better to remove prepared_content function from the plugin module?

Or just assert against a pinner output?

I am up for removing unused code

File: will/tests/test_plugin.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+import re
+
+import pytest
+from freezegun import freeze_time
+
+from will import settings
+from will.plugin import WillPlugin
+
+# Used to freeze_time
+WILLS_BIRTHDAY = "2013-12-04 01:11:00"
```
[the remark this answers is not in the document]
Comment (chillipeper) [reply will/tests/test_plugin.py]: ba93022

File: will/tests/test_plugin.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+
+@freeze_time(WILLS_BIRTHDAY)
+def test_reply_package_for_scheduling_is_true(plugin, content, message, event, io_backend):
+    original_message = message({"backend": io_backend})
+    incomming_event = event({"data": original_message})
+    backend = plugin.get_backend(original_message, None)
+    e = event({'type': "reply",
+               'content': content,
+               'topic': "message.outgoing.%s" % backend,
```
[the remark this answers is not in the document]
Comment (chillipeper) [reply will/tests/test_plugin.py]: 42c32e7

File: will/tests/test_plugin.py (hunk 1 above)
Comment (wontonst) [reply will/tests/test_plugin.py]: removing dead code seems like a good idea 👍 

any input, @rsalmond ?

**Comment by chillipeper** [comment]
@rsalmond @wontonst if you have some free time, could you review this PR.

Bumped coverage to 97%, there are only 4 lines of code that I think are not testable, and probably need to be removed. I added my comments on the code

Let me know what you think, and if this can be merged

**Comment by chillipeper** [comment]
For the sake of bumping up coverage will merge this one

**Pull request merged**